src/crimson/net/ProtocolV2.cc

```diff
@@ -1079,15 +1077,43 @@ ProtocolV2::send_wait()
   });
 }
 
+seastar::future<ProtocolV2::next_step_t>
+ProtocolV2::reuse_connection(
+    ProtocolV2* existing_proto, bool do_reset,
+    bool reconnect, uint64_t conn_seq, uint64_t msg_seq)
+{
+  existing_proto->trigger_replacing(reconnect,
+                                    do_reset,
+                                    std::move(socket),
+                                    std::move(auth_meta),
+                                    std::move(session_stream_handlers),
+                                    peer_global_seq,
+                                    client_cookie,
+                                    conn.get_peer_name(),
+                                    connection_features,
+                                    conn_seq,
+                                    msg_seq);
+  abort_in_fault();
```

Comment:
No, because we are reusing the existing Connection, this accepting Connection should end up with fault (which will be closed) after moving all necessary information to the existing Connection.

Reply:
so it's not a "fault" per-se. and what makes me feel nervous is that we are not using exception as way to describe "exception". is it possible to use `next_step_t` for conveying the message that we need to skip the following steps and need to do cleanup right away?

Reply:
there is `abort_in_close()`, maybe that is more clear?

Reply:
yeah, could you also accompany it with a comment explaining that we expect the error handling code to cleanup the existing connection when handling the thrown exception. probably we can revisit this part later to improve the readability.

Reply:
changed to `abort_in_close()` and added comments.